File: go-common/src/main/java/com/gjw/common/mapstruct/StudentMapper.java
package com.gjw.common.mapstruct;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.Mappings;
import org.mapstruct.factory.Mappers;

import java.util.List;

@Mapper
public interface StudentMapper {
    StudentMapper INSTANCE = Mappers.getMapper(StudentMapper.class);

    @Mappings({
            @Mapping(source = "name", target = "studentName"),
            @Mapping(source = "age", target = "studentAge")
    })
    StudentVO po2Vo(StudentPO studentPO);

    List<StudentVO> poList2VoList(List<StudentPO> studentPO);
}

File: src/main/kotlin/com/gjw/go/mapstruct/StudentMapper.kt
package com.gjw.go.mapstruct

import com.gjw.go.domain.entity.StudentPo
import com.gjw.go.excel.listener.student.StudentImpDto
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.factory.Mappers


/**
 * 参考地址：
 * https://github.com/mapstruct/mapstruct/issues/1672
 * https://github.com/mapstruct/mapstruct-examples
 */
@Mapper
interface StudentMapper {
    companion object {
        var INSTANCE = Mappers.getMapper(StudentMapper::class.java)
    }

    @Mapping(source = "schoolName", target = "school")
    @Mapping(source = "teacherName", target = "teacher")
    fun studentImpDtoToStudentPo(studentImpDto: StudentImpDto): StudentPo
}